### ISSUE TITLE
_events_char_array sometimes not initialized in scribble_cache

### ISSUE THREAD

**Issue by DragoniteSpam** [JujuAdams/scribble#90]
I'm not 100% sure where this originates from, but *sometimes* if I create a scribble with a unique occurrence name more than once `scribble_cache` will fail thanks to `_events_char_array` not being given a value. I can't pin down which line that would assign it a value is being skipped, although declaring

    var _events_char_array = [];

at the top of the script takes care of it.

Also, spell check tells me it's spelled "occurrence" instead of "occurance" but I don't know if I can be bothered to fix that in all of my code so.

**Comment by JujuAdams** [comment]
Thanks for the bug report

At line 58 in `scribble_cache()` you'll find
```GML
var _scribble_array = global.__scribble_global_cache_map[? _cache_string];
```

Underneath this line, please add:
```GML
_events_char_array = _scribble_array[SCRIBBLE.EVENT_CHAR_ARRAY];
```

Could you confirm whether this prevents the crash from occurring?

**Comment by DragoniteSpam** [comment]
That did it. Thank!

**Comment by JujuAdams** [comment]
Gonna reopen this so I remember to fix it in the repo 👀

**Comment by JujuAdams** [comment]
Hokay, fixed for v6.0.8
